lab3/client/src/reducer.js

```diff
@@ -0,0 +1,30 @@
+export default function reducer (state, action) {
+	switch (action.type) {
+	case 'EXAMPLE_MUTATION':
+		state.example = action.payload;
+		return state;
+	case 'BUTTON_CLICK':
+		console.log('Incremented coin by 1');
+		state.counter++;
+		return state;
+	case 'BUY_GENERATOR':
+
+		for (var i = 0; i < state.generators.length; i++) {
+
+			var newCost = Math.round((state.generators[i].baseCost*Math.pow(1 + 0.05, state.generators[i].quantity)).toFixed(2) * 100)/100;
```

Comment:
should probably reuse the implementation from the Generator. But I get what you are doing here.